src/commands/build.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{
+    commands::up::{UpCommand, UpOpts},
+    opts::{BUILD_UP_OPT, DEFAULT_MANIFEST_FILE},
+};
+use anyhow::Result;
+use spin_loader::local::{config::RawAppManifestAnyVersion, raw_manifest_from_file};
+use structopt::{clap::AppSettings, StructOpt};
+
+/// Run the build command for each component.
+#[derive(StructOpt, Debug)]
+#[structopt(
+    about = "Build the Spin application",
+    global_settings = &[AppSettings::ColoredHelp]
+)]
+pub struct BuildCommand {
+    #[structopt(flatten)]
+    pub opts: UpOpts,
+
+    /// Run the application after building.
+    #[structopt(name = BUILD_UP_OPT, short = "u", long = "up")]
+    pub up: bool,
+}
+
+impl BuildCommand {
+    pub async fn run(self) -> Result<()> {
+        let manifest_file = self
+            .opts
+            .app
+            .as_deref()
+            .unwrap_or_else(|| DEFAULT_MANIFEST_FILE.as_ref());
+        let RawAppManifestAnyVersion::V1(app) = raw_manifest_from_file(&manifest_file).await?;
+
+        spin_build::build(app, manifest_file).await?;
+
+        if self.up {
+            let cmd = UpCommand {
+                opts: UpOpts {
+                    address: self.opts.address,
+                    app: Some(manifest_file.into()),
```

Comment:
Does `app` need to be part of `UpOpts`? If it wasn't you could just `opts: self.opts` here.